plugins/modules/rudder_node_settings.py

```diff
@@ -174,9 +164,6 @@ def __init__(self, module):
         if module.params.get("rudder_url", None):
```

Comment:
You can put the default value here, instead of the "None". And then check if the value is localhost or not to force validate_certs if needed